src/server.c

```diff
@@ -1837,10 +1837,9 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
 
     /* Run a fast expire cycle (the called function will return
      * ASAP if a fast cycle is not needed). */
+    long long expire_cycle_time = 0;
     if (server.active_expire_enabled && !server.import_mode && iAmPrimary()) {
```

Comment:
Don't we need to skip the expire work if ProcessingEventsWhileBlocked, to make sure the active time is summed correctly?

```suggestion
    if (server.active_expire_enabled &&
        !server.import_mode &&
        !ProcessingEventsWhileBlocked &&
        iAmPrimary()) {
```